### PR TITLE
refactor ipam unit test

### PR DESCRIPTION
**Pull request by changluyi**
1) check all ipam and subnet module's export api
2) add pod with multi nic case
3) refactor comment in api By()
4) check more items in struct ipam
5) add case update err subnet cidr

- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

#### What type of this PR
Examples of user facing changes:
- Features
- Bug fixes
- Docs
- Tests

#### Which issue(s) this PR fixes:
Fixes #(issue-number)

### CODE REVIEW

File: test/unittest/ipam/ipam.go [hunk 1]
```diff
@@ -33,52 +35,101 @@ var _ = Describe("[IPAM]", func() {
...
 				im := ipam.NewIPAM()
+
+				By("invalid mask len > 32")
 				err := im.AddOrUpdateSubnet(subnetName, "1.1.1.1/64", v4Gw, nil)
 				Expect(err).Should(MatchError(ipam.ErrInvalidCIDR))
+
+				By("invalid ip range")
 				err = im.AddOrUpdateSubnet(subnetName, "1.1.256.1/24", v4Gw, nil)
 				Expect(err).Should(MatchError(ipam.ErrInvalidCIDR))
 			})
```
Comment (zhangzujian) [inline test/unittest/ipam/ipam.go:52]: Edge cases, such as prefix = 30/31/32/33/34, are needed.

File: test/unittest/ipam/ipam.go [hunk 2]
```diff
@@ -2,11 +2,14 @@ package ipam
 
 import (
 	"fmt"
+	"math/rand"
```
Comment (zhangzujian) [inline test/unittest/ipam/ipam.go:5]: The random source should be initialized.

File: test/unittest/ipam/ipam.go [hunk 3]
```diff
@@ -33,52 +36,103 @@ var _ = Describe("[IPAM]", func() {
 	dualCIDR := fmt.Sprintf("%s,%s", ipv4CIDR, ipv6CIDR)
 	dualExcludeIPs := append(ipv4ExcludeIPs, ipv6ExcludeIPs...)
 
+	// TODO test case use random ip and ipcidr, and input test data should seperate from test case
```
Comment (zhangzujian) [inline test/unittest/ipam/ipam.go:39]: typo: `separate`

File: test/unittest/ipam/ipam.go [hunk 4]
```diff
@@ -356,6 +532,13 @@ var _ = Describe("[IPAM]", func() {
 			Expect(ip2.Add(-1)).To(Equal(ip1))
 			Expect(ip1.Sub(-1)).To(Equal(ip2))
 			Expect(ip2.Sub(1)).To(Equal(ip1))
+
+			ipr := ipam.IPRange{Start: "10.0.0.1", End: "10.0.0.254"}
+			Expect(ipr.IPExist(ip1)).To(BeTrue())
+			Expect(ipr.IPExist(ip2)).To(BeTrue())
+
+			iprlist := ipam.IPRangeList{&ipr}
+			Expect(iprlist.Contains(ip1)).To(BeTrue())
```
Comment (zhangzujian) [inline test/unittest/ipam/ipam.go:541]: `iprlist` => `iprList`
